modules/gke-nodepool/variables.tf

```diff
@@ -158,10 +158,12 @@ variable "reservation_affinity" {
 variable "service_account" {
   description = "Nodepool service account. If this variable is set to null, the default GCE service account will be used. If set and email is null, a service account will be created. If scopes are null a default will be used."
   type = object({
-    email        = optional(string)
-    oauth_scopes = optional(list(string))
+    create       = optional(bool, false)
+    email        = optional(string, null)
+    oauth_scopes = optional(list(string), null)
   })
-  default = null
+  default  = {}
```

Comment:
What do you think `create=optional(bool, true)` and the `default={create=false}`

This way setting `service_account = {}` creates the service account. Currently you have to use `service_account = { create = true }` (see examples, and blueprints changes)

Reply:
yep, thanks for doing this got sidetracked by a meeting ;)

Reply:
Depends on what we want as a default: the implicit default till now was to use the default GCE SA, with the new format you would need `{create=false, email=null}` to achieve the same. Which kind of makes sense, as we would make it harder to do the "wrong" thing. Except on autopilot, where a pending provider bug ignores whatever you pass in and always uses the GCE default.

Reply:
it would also have the side effect of always requiring create=false for existing SAs, which is kind of counterintuitive; for me creation should be explicit, non-creation the default or you can very easily get into weird situations